lib/mayaUsd/ufe/UsdUndoInsertChildCommand.cpp

```diff
@@ -94,6 +90,12 @@ UsdUndoInsertChildCommand::UsdUndoInsertChildCommand(const UsdSceneItem::Ptr& pa
     _parentLayer = parentPrim.IsPseudoRoot() 
         ? parent->prim().GetStage()->GetEditTarget().GetLayer() 
         : MayaUsdUtils::defPrimSpecLayer(parentPrim);
+
+    // Apply restriction rules
+    if(!parentPrim.IsPseudoRoot()){
+        ufe::applyCommandRestriction(childPrim, "reparent");
+        ufe::applyCommandRestriction(parentPrim, "reparent");
+    }
```

Comment:
I was unconditionally applying the restriction rules here but realized that If parent prim is the pseudo-root, no def primSpec will be found and instead USD returns the "over".

This case happens when one tries to parent a prim under proxy shape ('/').